Add ShadCN UI Conditional Integration to create-next-app with Multi-Package Manager Support

vedantlavale opened the pull request:
# Add ShadCN UI Integration to create-next-app

## Overview
Integrates ShadCN UI setup directly into `create-next-app`, enabling automatic component library configuration during project creation with proper Tailwind CSS dependency handling.

## Changes
- **CLI Integration**: Added ShadCN prompt as optional feature (opt-in)
- **Dependency Management**: ShadCN prompt only appears when Tailwind CSS is enabled
- **Multi-Package Manager Support**:
  - **npm**: `npx shadcn@latest init`
  - **yarn**: `yarn dlx shadcn@latest init`
  - **pnpm**: `pnpm dlx shadcn@latest init`
  - **bun**: `bunx --bun shadcn@latest init`
- **New Helper**: Created `helpers/shadcn.ts` for cross-platform ShadCN initialization
- **Error Handling**: Graceful fallback if ShadCN setup fails
- **Smart Defaults**: Automatically disables ShadCN when Tailwind is not selected

## Files Modified
- `packages/create-next-app/index.ts` - Added CLI prompt, dependency logic, and display config
- `packages/create-next-app/create-app.ts` - Integrated installation workflow  
- `packages/create-next-app/helpers/shadcn.ts` - New ShadCN helper functions

## User Experience
**Before:**
```bash
npx create-next-app@latest my-app
cd my-app
npx shadcn@latest init  # Manual step required
```

**After:**
```bash
npx create-next-app@latest my-app
# ShadCN configured automatically when Tailwind is enabled ✨
```

## Dependency Logic
- **Tailwind Required**: ShadCN prompt only shows when Tailwind CSS is selected
- **Auto-disable**: If user selects "No" to Tailwind, ShadCN is silently disabled
- **CLI Support**: Handles `--no-tailwind` flag to disable both Tailwind and ShadCN
- **Clean UX**: No confusing prompts for incompatible configurations

## Benefits
- Eliminates manual ShadCN setup steps
- Prevents broken configurations (ShadCN without Tailwind)
- Ensures consistent component library configuration
- Maintains backward compatibility with opt-out option
- Supports all major package managers out of the box
- Smart dependency management for better user experience




##Screenshots
- Shows how ShadCN UI setup is conditionally presented only when Tailwind CSS is selected:
<img width="875" height="283" alt="Screenshot 2025-10-12 at 4 56 51 PM" src="https://github.com/user-attachments/assets/7a677844-8597-4854-862d-0e03fbb378b5" />

- Visualizes install process including ShadCN setup and all major package manager options:
<img width="1033" height="874" alt="Screenshot 2025-10-12 at 4 58 00 PM" src="https://github.com/user-attachments/assets/a953841b-1065-4496-887e-02f410270d70" />

vedantlavale committed:
add ShadCN UI installation option and helper function

vedantlavale committed:
enable ShadCN integration by default in create-next-app in

vedantlavale committed:
ensure ShadCN integration is contingent on Tailwind CSS being enabled

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: 9d9fdcd81aa0a94ac8c59cb27fe0f45dea5e83c8

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

vedantlavale committed:
enhance ShadCN installation prompt with cancellation option

vedantlavale commented:
<img width="1000" height="701" alt="Screenshot 2025-10-12 at 6 00 52 PM" src="https://github.com/user-attachments/assets/3d492df9-e70a-4024-b5ae-b256cbc2343b" />
 Handle user cancellation during ShadCN setup.
 If Ctrl+C is pressed, set flag to skip ShadCN install.
 Next.js app setup continues normally, but ShadCN installation is skipped.
 Shows a message informing users they can add ShadCN later manually.

vedantlavale committed:
fix naming

vedantlavale committed:
support all package managers for ShadCN installation

Auto-detect Yarn version and run appropriate install commands to ensure
ShadCN uses the correct package manager (npm/yarn/pnpm/bun)
